landlab/components/profiler/trickle_down_profiler.py

```diff
@@ -0,0 +1,392 @@
+           [  8.,   8.,   4.,   3.,   2.,   0.,   3.,   0.,   0.,   0.],
+           [  0.,   3.,   0.,   0.,   1.,   0.,   2.,   0.,   0.,   0.],
+           [  0.,   2.,   1.,   0.,   0.,   0.,   1.,   0.,   0.,   0.],
+           [  0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.,   0.]])
+
+    >>> profiler = TrickleDownProfiler(
+    ...     mg,
+    ...     starting_nodes=[54, 66])
+    >>> profiler.run_one_step()
+
+    The keys of the property ``data_structure`` are the IDs  of the two
+    outlet nodes.
+
+    >>> profiler.data_structure.keys()
+    odict_keys([54, 66])
+
+    Within the data structure, the value at key 54, is a dictionary of the
+    one segment, each specified by a ``(dowstream, upstream)`` tuple:
+
+    >>> profiler.data_structure[54].keys()
+    dict_keys([(40, 54)])
+
+    The value of the segment between nodes 40 and 54 has the following
+    components:
+
+    >>> profiler.data_structure[54][(40, 54)]["ids"]
+    array([40, 41, 42, 43, 44, 54])
+    >>> profiler.data_structure[54][(40, 54)]["distances"]
+    array([ 0.,  1.,  2.,  3.,  4.,  5.])
+    >>> np.round(profiler.data_structure[54][(40, 54)]["color"], decimals=2)
+    array([ 0.27,  0.  ,  0.33,  1.  ])
+
+    The rest of the ``profile_structure`` encodes information about the second
+    profile which starts at node 66.
+
+    >>> profiler.data_structure[66].keys()
+    dict_keys([(8, 66)])
+
+    >>> profiler.data_structure[66][(8, 66)]["ids"]
+    array([ 8, 18, 17, 16, 26, 36, 46, 56, 66])
+    >>> profiler.data_structure[66][(8, 66)]["distances"]
+    array([ 0.,  1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.])
+    >>> np.round(profiler.data_structure[66][(8, 66)]["color"], decimals=2)
+    array([ 0.13,  0.57,  0.55,  1.  ])
+
+
+    References
+    ----------
+    **Required Software Citation(s) Specific to this Component**
+
+    None Listed
+
+    **Additional References**
+
+    None Listed
+
+    """
+
+    _name = "TrickleDownProfiler"
+
+    _unit_agnostic = True
+
+    _info = {
+        "flow__link_to_receiver_node": {
+            "dtype": int,
+            "intent": "in",
+            "optional": False,
+            "units": "-",
+            "mapping": "node",
+            "doc": "ID of link downstream of each node, which carries the discharge",
+        },
+        "flow__receiver_node": {
+            "dtype": int,
+            "intent": "in",
+            "optional": False,
+            "units": "-",
+            "mapping": "node",
+            "doc": "Node array of receivers (node that receives flow from current node)",
+        },
+    }
+
+    def __init__(
+        self, grid, starting_nodes=None, cmap="viridis",
+    ):
+        """
+        Parameters
+        ----------
+        grid : Landlab Model Grid instance, required
+        starting_nodes : iterable
+        cmap : str
+            A valid matplotlib cmap string. Default is "viridis".
+
+        """
+        super(TrickleDownProfiler, self).__init__(grid)
```

Comment:
Until now I didn't know anything about the new super (but had found that error...) I suspect this is an old holdover. 

@mcflugen any comments? 

I'm a glutton for consistency (and always like practicing regex, so I just changed it everywhere)... tests pass locally.

Reply:
@kbarnhart I agree this is just a holdover from Python 2.7. If we're going to use *super*, we should probably use the new form since we're all Python 3 now.

Reply:
Glad you think so @mcflugen. Its now all changed :)